shopfloor_base_default_profile/models/shopfloor_app_user_profile.py

```diff
@@ -0,0 +1,11 @@
+# Copyright 2022 Brian McMaster <brian@mcmpest.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+from odoo import fields, models
+
+
+class ShopfloorAppUserProfile(models.Model):
+    _name = "shopfloor.app.user.profile"
```

Comment:
Do you really need this model?
Since profiles are linked to the app already https://github.com/OCA/wms/blob/14.0/shopfloor_base/models/shopfloor_app.py#L73 you can simply have a m2m on the user.